### PR TITLE
Intt survey geo

### PR DESCRIPTION
**Pull request by hrjheng**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)
Implement INTT survey geometry in PHG4InttDetector. The corresponding PR for the macro: https://github.com/sPHENIX-Collaboration/macros/pull/790
[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

**Comment by osbornjd** [comment]
Here's another build with the macros PR too
https://web.sdcc.bnl.gov/jenkins-sphenix/job/sPHENIX/job/Build-Master-gcc12/2535/

**Comment by hrjheng** [comment]
The attached plots show the phi and eta distributions of TrkrCluster and PHG4Particle, from a simulation of 1000 events of 100 pions.  

clusEta_histogram
clusPhi_histogram
PHG4PEta_histogram
PHG4PPhi_histogram

File: offline/packages/trackbase/ActsSurfaceMaps.cc [hunk 1]
```diff
@@ -74,6 +74,8 @@ Surface ActsSurfaceMaps::getSiliconSurface(TrkrDefs::hitsetkey hitsetkey) const
   unsigned int trkrid = TrkrDefs::getTrkrId(hitsetkey);
   TrkrDefs::hitsetkey tmpkey = hitsetkey;
 
+  std::cout << "trkrid = " << trkrid << ", TrkrDefs::inttId = " << TrkrDefs::inttId << ", TrkrDefs::mvtxId = " << TrkrDefs::mvtxId << std::endl;
```
Comment (blackcathj) [inline offline/packages/trackbase/ActsSurfaceMaps.cc:77]: Please suppress these `cout`s under a verbosity check

Comment (hrjheng) [reply offline/packages/trackbase/ActsSurfaceMaps.cc]: Done

**Comment by osbornjd** [comment]
The macros are failing to run because for some reason the INTT geometry is not found by the Acts geometry builder:
https://web.sdcc.bnl.gov/jenkins-sphenix/job/sPHENIX/job/test-default-detector-pipeline/7367/console
Obviously this must not be the case since you were able to run locally, so is there some change you may have had locally that did not get propagated to the branch?

**Comment by cdean-github** [comment]
> The macros are failing to run because for some reason the INTT geometry is not found by the Acts geometry builder: https://web.sdcc.bnl.gov/jenkins-sphenix/job/sPHENIX/job/test-default-detector-pipeline/7367/console Obviously this must not be the case since you were able to run locally, so is there some change you may have had locally that did not get propagated to the branch?

Hi @osbornjd,

Looking at this report, the CDB file it's loading is from Joseph Bertaux's local area. Is it possible that the file has yet to be loaded to the CDB?
```
Use the survey geometry. Get the survey map from CDB/file
int InttSurveyMap::LoadFromFile(std::string const& filename)
	File '/sphenix/u/jbertaux/sphnx_software/INTT/general_codes/josephb/codes/intt_alignment/dat/intt_survey_cdbttree.root' does not exist
```

**Comment by osbornjd** [comment]
Good eye, I was hasty in just skipping down to the Acts error and assuming that was the cause of the problem. That would explain why it works on gpfs without a problem 😄 . @pinkenburg  can you put '/sphenix/u/jbertaux/sphnx_software/INTT/general_codes/josephb/codes/intt_alignment/dat/intt_survey_cdbttree.root' into the CDB for us? Then I can help @hrjheng to update the PR to pull from the CDB instead of from this absolute path on gpfs

**Comment by hrjheng** [comment]
Note for record: I expect a few changes and modifications to be implemented after [Joseph's PR](https://github.com/sPHENIX-Collaboration/coresoftware/pull/2608) is merged

**Comment by pinkenburg** [comment]
the valgrind error is from a shift in line numbers in the PHG4InttDetector.cc, the updated suppression file will take care of that

**Pull request merged**